### ISSUE TITLE
Dropping coins with 'D', '$', in traditional menu asks to drop everything. 

### ISSUE THREAD

**Issue by qpl23** [k21971/EvilHack#61]
I've been having this problem (playing on hardfought, .evilrc: https://www.hardfought.org/userdata/m/mung/evilhack/mung.evilrc )

Using traditional menustyle, if I press 'D' to drop multiple items, then '$' and Enter to drop coins, I get asked if I want to drop my coins, but:

1. If I say 'y', it will then cycle through all other items asking me if I want to drop them.
2. If I say 'a', it drops all of my inventory.

I spent a bit of time with the code, and I've written a fix to illustrate where the problem is, but it may break other things - I've only tested to make sure it stops the above behaviour.

Essentially, in askchain for the COIN_CLASS case, '$' is not getting added to olets, so (I think) it is as if player simply pressed Enter rather than '$', Enter. Dumb fix follows:

```
diff --git a/src/invent.c b/src/invent.c
index 43bb11c38..f03ad257e 100644
--- a/src/invent.c
+++ b/src/invent.c
@@ -2090,6 +2090,11 @@ unsigned *resultflags;
 
         if (oc_of_sym == COIN_CLASS && !combo) {
             context.botl = 1;
+            if (!index(olets, oc_of_sym)) {
+                add_valid_menu_class(oc_of_sym);
+                olets[oletct++] = oc_of_sym;
+                olets[oletct] = 0;
+            }
         } else if (sym == 'a') {
             allflag = TRUE;
         } else if (sym == 'A') {
```

**Comment by entrez** [comment]
I think you can just drop the entire `if (oc_of_sym == COIN_CLASS && !combo)` part entirely, so that the if/else chain starts with `if (sym == 'a')`.  That way `$` will be handled by the default case where `add_valid_menu_class(oc_of_sym)` and that other stuff is already done for other object classes.  It looks like `context.botl` is already set to 1 by `freeinv_core` when you are dropping gold, so doing the same thing in `ggetobj` is probably not necessary?

**Comment by k21971** [comment]
Closed via git commit fcdcd27.
